File: src/frontend/src/types/system-design.ts
export interface SystemDesignSummary {
  id: number;
  slug: string;
  title: string;
  subtitle: string | null;
  diagram_filename: string | null;
  display_order: number;
}

export interface SystemDesign extends SystemDesignSummary {
  overview: string | null;
  architecture: string | null;
  dataflow: string | null;
  formulas: string | null;
  production_constraints: string | null;
  tradeoffs: string | null;
  defense: string | null;
  verbal_outline: string | null;
  cheat_sheet: string | null;
  created_at: string;
  updated_at: string;
}

export interface SystemDesignCheatSheet extends SystemDesignSummary {
  cheat_sheet: string | null;
}

export type SystemDesignSection =
  | "overview"
  | "architecture"
  | "dataflow"
  | "formulas"
  | "production_constraints"
  | "tradeoffs"
  | "defense"
  | "verbal_outline"
  | "cheat_sheet";

export const SECTION_LABELS: Record<SystemDesignSection, string> = {
  overview: "概览 (Overview)",
  architecture: "架构 (Architecture)",
  dataflow: "数据流 (Data Flow)",
  formulas: "公式估算 (Formulas)",
  production_constraints: "生产约束 (Production Constraints)",
  tradeoffs: "权衡取舍 (Tradeoffs)",
  defense: "应答策略 (Defense)",
  verbal_outline: "口述脉络 (Verbal Outline)",
  cheat_sheet: "速查表 (Cheat Sheet)",
};

File: src/backend/models/system_design.py
"""System design case study model."""
from datetime import datetime, timezone

from sqlalchemy import Column, DateTime, Integer, String, Text

from src.backend.database import Base


class SystemDesign(Base):
    """A system design case study module with 8 markdown content sections."""

    __tablename__ = "system_designs"

    id = Column(Integer, primary_key=True, autoincrement=True)
    slug = Column(String(100), unique=True, nullable=False, index=True)
    title = Column(Text, nullable=False)
    subtitle = Column(Text, nullable=True)
    diagram_filename = Column(String(255), nullable=True)

    # 8 markdown content sections (all editable via frontend)
    overview = Column(Text, nullable=True)                # S1: Overview & Motivation
    architecture = Column(Text, nullable=True)            # S2: Architecture Deep Dive
    dataflow = Column(Text, nullable=True)                # S3: Data Flow & Key Components
    formulas = Column(Text, nullable=True)                # S4: Formulas & Algorithms
    production_constraints = Column(Text, nullable=True)  # S5: Production Constraints
    tradeoffs = Column(Text, nullable=True)               # S6: Trade-off Analysis
    defense = Column(Text, nullable=True)                 # S7: Adversarial Defense Q&A
    verbal_outline = Column(Text, nullable=True)          # S8: Verbal Outline (3-min & 10-min)

    display_order = Column(Integer, default=0)
    created_at = Column(DateTime, default=lambda: datetime.now(timezone.utc))
    updated_at = Column(
        DateTime,
        default=lambda: datetime.now(timezone.utc),
        onupdate=lambda: datetime.now(timezone.utc),
    )
